core/src/test/java/org/apache/iceberg/rest/requests/TestCreateTableRequest.java

```diff
@@ -0,0 +1,293 @@
+    );
+
+    AssertHelpers.assertThrows(
+        "The builder should not allow using null for the schema",
+        NullPointerException.class,
+        "Invalid schema: null",
+        () -> CreateTableRequest.builder().withSchema(null)
+    );
+
+    AssertHelpers.assertThrows(
+        "The builder should not allow passing a null collection of properties",
+        NullPointerException.class,
+        "Invalid collection of properties: null",
+        () -> CreateTableRequest.builder().setProperties(null)
+    );
+
+    Map<String, String> mapWithNullKey = Maps.newHashMap();
+    mapWithNullKey.put(null, "hello");
+    AssertHelpers.assertThrows(
+        "The builder should not allow using null as a key in the properties to set",
+        IllegalArgumentException.class,
+        "Invalid property: null",
+        () -> CreateTableRequest.builder().setProperties(mapWithNullKey)
+    );
+
+    Map<String, String> mapWithNullValue = Maps.newHashMap();
+    mapWithNullValue.put("a", null);
+    mapWithNullValue.put("b", "b");
+    AssertHelpers.assertThrows(
+        "The builder should not allow using null as a value in the properties to set",
+        IllegalArgumentException.class,
+        "Invalid value for properties [a]: null",
+        () -> CreateTableRequest.builder().setProperties(mapWithNullValue).build()
+    );
+
+    AssertHelpers.assertThrows(
+        "The builder should not allow using null as a value when setting a single property",
+        IllegalArgumentException.class,
+        "Invalid value for property foo: null",
+        () -> CreateTableRequest.builder().setProperty("foo", null)
+    );
+
+    AssertHelpers.assertThrows(
+        "The builder should not allow using null as a key when setting a single property",
+        IllegalArgumentException.class,
+        "Invalid property: null",
+        () -> CreateTableRequest.builder().setProperty(null, "foo")
+    );
+  }
+
+  @Override
+  public String[] allFieldsFromSpec() {
+    return new String[] {"name", "location", "schema", "spec", "order", "properties", "stageCreate"};
+  }
+
+  @Override
+  public CreateTableRequest createExampleInstance() {
+    return CreateTableRequest.builder()
+        .withName(SAMPLE_NAME)
+        .withLocation(SAMPLE_LOCATION)
+        .withSchema(SAMPLE_SCHEMA)
+        .withPartitionSpec(SAMPLE_SPEC)
+        .withWriteOrder(SAMPLE_WRITE_ORDER)
+        .setProperties(SAMPLE_PROPERTIES)
+        .stageCreate()
+        .build();
+  }
+
+  @Override
+  public void assertEquals(CreateTableRequest actual, CreateTableRequest expected) {
+    Assert.assertEquals("Name should be the same", expected.name(), actual.name());
+    Assert.assertEquals("Location should be the same if provided", expected.location(), actual.location());
+    Assert.assertTrue("Schemas should be equivalent and have same schema id",
+        expected.schema().sameSchema(actual.schema()) && expected.schema().schemaId() == actual.schema().schemaId());
```

Comment:
I originally had this as two assertions (one for `sameSchema` and one for equality checking on `schemaId`).

If we prefer, I can revert to that.